### PR TITLE
VM speedup by using in-place bn.js operators

### PR DESCRIPTION
**Pull request by axic**
Replaces #37.

### CODE REVIEW

**Comment by axic** [comment]
@cdetrio this could be taken into consideration when testing is working again

**Comment by axic** [comment]
This one could be merged if the tests pass. There should be a slight speed increase.

**Comment by axic** [comment]
It fails some tests, should be fixed.

File: lib/opFns.js [hunk 1]
```diff
@@ -15,99 +15,97 @@ module.exports = {
...
   DIV: function (a, b, runState) {
-    a = new BN(a)
     b = new BN(b)
 
     var r
     if (b.isZero()) {
       r = [0]
     } else {
-      r = a.div(b).toArray('be', 32)
+      r = new BN(a).idiv(b).toArray('be', 32)
```
Comment (axic) [inline lib/opFns.js]: Useless change.

**Comment by axic** [comment]
Fixed finally, should pass all tests.

File: lib/opFns.js [hunk 2]
```diff
@@ -16,99 +16,99 @@ module.exports = {
...
   DIV: function (a, b, runState) {
-    a = new BN(a)
     b = new BN(b)
 
     var r
     if (b.isZero()) {
       r = [0]
     } else {
+      a = new BN(a)
       r = a.div(b).toArray('be', 32)
```
Comment (gasolin) [inline lib/opFns.js]: can do `r = new BN(a).div(b).toArray('be', 32)`

File: lib/opFns.js [hunk 3]
```diff
@@ -16,99 +16,99 @@ module.exports = {
...
   SDIV: function (a, b, runState) {
-    a = utils.fromSigned(a)
     b = utils.fromSigned(b)
 
     var r
     if (b.isZero()) {
       r = Buffer.from([0])
     } else {
+      a = utils.fromSigned(a)
       r = utils.toUnsigned(a.div(b))
```
Comment (gasolin) [inline lib/opFns.js]: can use `r = utils.toUnsigned(utils.fromSigned(a).div(b))`

File: lib/opFns.js [hunk 4]
```diff
@@ -16,99 +16,99 @@ module.exports = {
...
   },
   MOD: function (a, b, runState) {
-    a = new BN(a)
     b = new BN(b)
     var r
     if (b.isZero()) {
       r = [0]
     } else {
+      a = new BN(a)
       r = a.mod(b).toArray('be', 32)
```
Comment (gasolin) [inline lib/opFns.js]: `r = new BN(a).mod(b).toArray('be', 32)`

File: lib/opFns.js [hunk 5]
```diff
@@ -16,99 +16,99 @@ module.exports = {
...
   SMOD: function (a, b, runState) {
-    a = utils.fromSigned(a)
     b = utils.fromSigned(b)
     var r
 
     if (b.isZero()) {
       r = Buffer.from([0])
     } else {
+      a = utils.fromSigned(a)
       r = a.abs().mod(b.abs())
```
Comment (gasolin) [inline lib/opFns.js]: `r = utils.fromSigned(a).abs().mod(b.abs())`

File: lib/opFns.js [hunk 6]
```diff
@@ -16,99 +16,99 @@ module.exports = {
...
   ADDMOD: function (a, b, c, runState) {
-    a = new BN(a).add(new BN(b))
     c = new BN(c)
     var r
 
     if (c.isZero()) {
       r = [0]
     } else {
+      a = new BN(a).iadd(new BN(b))
       r = a.mod(c).toArray('be', 32)
```
Comment (gasolin) [inline lib/opFns.js]: same here

File: lib/opFns.js [hunk 7]
```diff
@@ -16,99 +16,99 @@ module.exports = {
...
   MULMOD: function (a, b, c, runState) {
-    a = new BN(a).mul(new BN(b))
     c = new BN(c)
     var r
 
     if (c.isZero()) {
       r = [0]
     } else {
+      a = new BN(a).imul(new BN(b))
       r = a.mod(c).toArray('be', 32)
```
Comment (gasolin) [inline lib/opFns.js]: same here

File: lib/opFns.js (hunk 2 above)
Comment (axic) [inline lib/opFns.js]: All this was intentional to keep the change diff small and reviewable. It is going to be replaced by #159 anyway.

**Pull request merged**